frontend/src/ComponentConfigNode.svelte

```diff
@@ -0,0 +1,360 @@
+    box-shadow: inset 2px 2px 0 var(--main-fg-color),
+      inset -2px -2px 0 var(--main-fg-color);
+  }
+
+  .container {
+    min-height: 100%;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: center;
+    align-items: stretch;
+    align-content: center;
+  }
+
+  label {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: flex-start;
+    align-items: baseline;
+  }
+
+  input {
+    background: inherit;
+    color: inherit;
+    border: none;
+    border-bottom: 1px solid var(--main-fg-color);
+    flex-grow: 1;
+    margin-left: 1ch;
+    /* width: 100%;
+    margin: 0;
+    padding: 0; */
+  }
+
+  input:invalid {
+    border-bottom: 1px solid var(--comment-color);
+  }
+
+  input:focus {
+    outline: none;
+    box-shadow: inset 0 -1px 0 var(--main-fg-color);
+  }
+
+  input:focus:invalid {
+    outline: none;
+    box-shadow: inset 0 -1px 0 var(--comment-color);
+  }
+
+  .input {
+    padding-left: 2em;
+    padding-bottom: 0.5em;
+    border-left: 1px dashed white;
+  }
+
+  select {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    padding: 0.5em 1em;
+    margin-left: 0.5em;
+    margin-right: 0.5em;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  select {
+    width: 100%;
+    margin: 0;
+    margin-bottom: 1em;
+  }
+
+  option {
+    font-family: monospace;
+  }
+
+  .boxed {
+    border: 2px solid var(--main-fg-color);
+    padding: 2em;
+    margin: 1em 0 2em 0;
+  }
+
+  .buttonbar {
+    top: 0;
+    left: 0;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    align-items: start;
+    justify-content: start;
+  }
+
+  .buttonbar .remove {
+    display: flex;
+    align-items: start;
+    justify-content: start;
+    flex-direction: row;
+  }
+
+  .buttonbar .add {
+    display: flex;
+    align-items: start;
+    justify-content: start;
+    flex-direction: row;
+  }
+</style>
+
+<script>
+  import Checkbox from "./Checkbox.svelte";
+  import { calculator, splitAndCapitalize } from "./helpers";
+  import Icon from "./Icon.svelte";
+  import FileBrowser from "./FileBrowser.svelte";
+  export let node, element;
+  let unionTypeSelect,
+    _element,
+    intInput,
+    files = null;
+
+  const INT_PLACEHOLDERS = [
+    "0x10 * 2 + 8",
+    "(0x10 * 0x100 + 25) * 69",
+    "0x3 + 5 - 0x2",
+    "6 * 0x2 + 1",
+    "0x4 * (0x2 + 2)",
+    "12 / (0x3 * 0x2)",
+    "0x7 - 0x2 * 0x3",
+    "0x2 * (0x3 + 3)",
+    "(0x2 ^ 3) - 1",
+    "10 + 0x3 * 2",
+    "0x3 * (0x2 + 2) + 2",
+    "0x7 - 1 + 10",
+  ];
+
+  // $: if (node["type"] == "builtins.int") {
+  //   try {
+  //     element = calculator.calculate(_element);
+  //     intInput?.setCustomValidity("");
+  //   } catch {
+  //     element = undefined;
+  //     intInput?.setCustomValidity("Invalid expression.");
+  //   }
+  // }
```

Comment:
Why is this commented it out? If it's causing a problem, we should probably address it a different way.

Reply:
I'm not sure of the root reason, but when clearing an element of a list or dict with an int value, the value that was being cleared would propagate to the the new nth element in the list/dict.

For instance if i try to remove element 2 of [(n1, k1), (n2,k2), (n3,k3)] where n is an int it will become  [(n1, k1), (n2,k3)].

I think it has something to do with the reactivity of the calculator line, but was kind of difficult to figure out.